### PR TITLE
Add exact Grok, GroveMoE, and Hunyuan-MoE GGUF graphs

### PR DESCRIPTION
**Pull request by justinchuby**
## Summary

- add dedicated float-import GGUF graphs for `grok`, `grovemoe`, and `hunyuan-moe`
- close pinned metadata defaults, conditional tensor families, fused/split QKV layouts, suffix-exact mappings, and packed expert expansion
- preserve architecture-specific semantics: Grok scaling/softcaps/sandwich norms/GEGLU, GroveMoE two-bank routing, and Hunyuan post-RoPE Q/K norm plus parallel shared expert
- keep quantized/native import rejected and runtime packaging deferred until independent real-weight and OGA evidence exists
- update model registry, route census, model coverage, and generated GGUF documentation

## Upstream and artifact evidence

Implementation is pinned to `ggml-org/llama.cpp@8d9af256337d1a501250f9bbf4c0859a654bddd6`.

Bounded HTTP range reads validated 8 MiB from each immutable artifact (24 MiB total; payloads removed afterward):

| Architecture | Artifact | Revision | Header result |
|---|---|---|---|
| `grok` | `unsloth/grok-2-GGUF/Q2_K/grok-2-Q2_K-00001-of-00003.gguf` | `7b560aeb31bbd8ff7a026f3cfd981c0364024cbf` | `grok`, 477 tensors, split 0/3, 963 logical tensors |
| `grovemoe` | `mradermacher/GroveMoE-Inst-GGUF/GroveMoE-Inst.Q2_K.gguf` | `4dec2e9eaf01d5150d5b45a7d7c95c6220aaad7f` | `grovemoe`, 723 tensors |
| `hunyuan-moe` | `gabriellarson/Hunyuan-A13B-Instruct-GGUF/Hunyuan-A13B-Instruct-IQ1_S.gguf` | `1ab8a37a314dfa05d4dc11cd899a5dded62c0578` | `hunyuan-moe`, 482 tensors |

The older Arki05 Grok-1 IQ1_S header was also audited and deliberately rejected as a current route because it uses the legacy per-expert `ffn_{gate,up,down}.N.weight` layout rather than the pinned loader's packed `*_exps` tensors.

GroveMoE intentionally follows the pinned llama.cpp graph where the chunk bank consumes the primary MoE output and gathers weights at floor-mapped chunk IDs, even though those details differ from the Hugging Face reference.

## Validation

- broad serial non-integration suite: 8,916 passed, 64 skipped, 12 deselected
- focused executable synthetic parity and end-to-end GGUF builds for all three architectures
- build-graph and weight-alignment coverage for all three internal model types
- lintrunner formatter and full lint clean
- GGUF support docs, model docs, and feature-flag generators clean
- diff check clean
- changed-line mypy: 0 errors across 10 changed production files
- two independent high-effort reviews: no significant findings

## Capability boundaries / waivers

- **Quantized/native storage:** not claimed. `keep_quantized=True` fails closed because these dedicated graphs currently expose floating `Linear`/expert modules only.
- **Runtime/OGA:** deferred. No full representative artifact plus independent ORT GenAI prefill, cached-decode, and generation evidence was established.
- **Real-weight golden generation:** deferred with runtime evidence. Grok has no public logical GGUF at or below 16 GiB; the only sub-16-GiB Hunyuan artifact is highly lossy IQ1_S. These limitations do not weaken the synthetic executable parity and exact graph-import acceptance gate added here.

### CODE REVIEW

File: src/mobius/integrations/gguf/_remaining_moe.py [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+    eps = float(metadata[f"{architecture}.attention.layer_norm_rms_epsilon"])
+    if not math.isfinite(eps) or eps <= 0:
+        raise ValueError(f"{architecture} GGUF has invalid normalization epsilon")
+
+    vocab = _vocab_size(metadata, architecture)
+    actual = {
+        name: tuple(int(dimension) for dimension in shape)
+        for name, _raw, _qtype, shape in gguf_model.tensor_items_raw()
+        if not is_known_skip(name)
+    }
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/integrations/gguf/_remaining_moe.py:182]: Fixed in `65df276a`. The core architecture closure now filters only suffix-valid mapped `.scale`/`.input_scale` sidecars, leaving them to the existing auxiliary-quantization preflight. That preflight still fails closed because these float-only graphs cannot represent the sidecars; unowned sidecars remain rejected. Added tests for both the mapped fail-closed path and an unexpected unowned sidecar.

File: src/mobius/models/moe.py [hunk 2]
```diff
@@ -451,6 +454,323 @@ def preprocess_weights(
...
+        position_ids: ir.Value,
+        past_key_values: list | None = None,
+    ):
+        logits, present_key_values = super().forward(
+            op,
+            input_ids,
+            attention_mask,
+            position_ids,
+            past_key_values,
+        )
```
[the remark this answers is not in the document]
Comment (justinchuby) [reply src/mobius/models/moe.py]: Fixed in `65df276a`. The Grok wrapper now captures the parent result once, scales only `outputs[0]`, and returns `(logits, *outputs[1:])`, preserving every optional base-model output instead of assuming a two-item tuple. A regression test enables `output_final_hidden_state` and verifies the `mtp_seed` output is retained.
